src/Compilers/CSharp/Test/Emit/CodeGen/FixedSizeBufferTests.cs

```diff
@@ -14,6 +14,73 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests.CodeGen
 {
     public class FixedSizeBufferTests : EmitMetadataTestBase
     {
+        [Fact]
+        public void NestedStructFixed()
+        {
+            var verifier = CompileAndVerify(@"
+class  X {
+    internal struct CaseRange {
+        public int Lo, Hi;
+        public unsafe fixed int Delta [3];
+
+        public CaseRange (int lo, int hi, int d1, int d2, int d3)
+        {
+            Lo = lo;
+            Hi = hi;
+            unsafe {
+                fixed (int *p = Delta) {
```

Comment:
>fixed (int *p = Delta) { [](start = 16, length = 24)

Consider adding a test for a static fixed field and for access to an instance fixed field outside of the declaring type.

Reply:
Static fixed fields are tested in FixedSizeBufferTests.StaticFixedBuffer and instance fixed fields outside the declaring type is tested in FixedSizeBufferTests.FixedStatementOnFixedBuffer.